docs/proposals/20200506-conditions.md

```diff
@@ -0,0 +1,363 @@
+        - [Beta -> GA Graduation](#beta---ga-graduation)
+        - [Removing a deprecated flag](#removing-a-deprecated-flag)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks [optional]](#drawbacks-optional)
+  - [Alternatives [optional]](#alternatives-optional)
+  - [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+Condition: The state of an object with regard to its appearance, quality, or working order.
+
+## Summary
+
+In Cluster API a workload cluster is composed of a fair number of Kubernetes objects, and to
+understand the status of your Cluster you have to jump from one object to another. Or, in the
+worst case, you have to look at the logs from the Cluster API pods to get more details on why
+things aren’t working.
+
+This proposal introduces the concept of “conditions” as a tool designed to provide “at glance”
+view of the status of a cluster, providing immediate answers and fine-grained guidance when
+investigating issues. 
+
+## Motivation
+
+- Cluster API requires a large number of objects to define a workload cluster: Cluster,
+  InfrastructureCluster, KubeadmControlPlane, MachineDeployments, Machines, and so on.
+  Each object carries its own status. Operators need to describe each object individually to
+  determine an overall cluster health summary.
+- Objects can be tied to each other directly (via object references), or indirectly
+  (via owner references). These dependencies can often require a deep knowledge of the
+  project’s internals and its APIs to investigate issues or problems. 
+- Some objects, like MachineSets or MachineDeployments, are considered a collection of
+  other objects (e.g. MachineSets). Their status is effectively tied to the status of each
+  item in the collection. 
+
+### Goals
+
+In order to make this proposal actionable in the v0.3.x timeframe, the proposal is going to
+be developed in an incremental fashion in three iterations. 
+
+The goal of this first iteration is:
+
+1. To agree on a design for the condition types.
+2. To provide backward-compatible support for `Phase`, `ErrorMessage`, `ErrorReason` fields.
+3. To provide utilities and libraries for a unified way to work with conditions.
+4. To provide flexibility for providers to incrementally adopt this proposal; or add, remove,
+   or update their own set of conditions.
+
+Once there will be an agreed design for the condition types, a second iteration will start
+with the following goals:
+
+1. To define the first set of conditions on the different Cluster API objects.
+2. To have `clusterctl describe cluster` present an overview of the cluster, its infrastructure,
+  control plane, and other related objects.
```

Comment:
Shouldn't these to be part of the first iteration?

Reply:
Considered the fact that we are anticipating the implementation of the process for making conditions visible thought the hierarchy, I'm putting this goal on hold